SCClassLibrary/Common/Control/Buffer.sc

```diff
@@ -595,6 +601,17 @@ Buffer {
 		^[\b_query, bufnum]
 	}
 
+	setSampleRate { arg value;
+		// sends the message first to check sanity of sr before changing the langage-side instance variable
+		server.listSendMsg(this.setSampleRateMsg(value));
+		sampleRate = value.asFloat;
+    }
+
+	setSampleRateMsg { arg value;
+		if(value.isNil) { Error("Cannot change the sample rate of a % that has been freed".format(this.class.name)).throw };
```

Comment:
I don't understand this error message...

Reply:
I'll make it readable

Reply:
Don't forget about the error message :)